src/prototype_creep_resources.js

```diff
@@ -726,9 +726,18 @@ Creep.prototype.construct = function() {
 };
 
 Creep.prototype.getTransferTargetStructure = function() {
-  const structure = this.pos.findClosestStructureWithMissingEnergyByRange(
+  let structure = this.pos.findClosestStructureWithMissingEnergyByRange(
     (object) => object.structureType !== STRUCTURE_STORAGE,
   );
+  // Harvester should always prefer Spawn and Extensions
```

Comment:
Harvester only uses this method before the storage is built, so the only structures are `SPAWN`, `EXTENSION`, and `TOWER`. Because of invaders and early attacks, I think it is a good idea to have the `TOWER` filled, too.

As soon as the storage is built up (and has enough energy) it will move on a fixed path and only fills spawns and extensions. I could only see issues at that point when also other structures are available.

So fine leave it in right now, but needs to be verified in early scenarios regarding the tower.

Reply:
I had a problem while all buildings were done. There was just an harvester from the previous bug and this harvester only filled the `TERMINAL`.
The tower would be filled after the room is working smootly again don't know if that brings a problem in the early stages.

Reply:
Ok, let's leave it like this. I don't like special cases so much, but better be robust for now.